### DISCUSSION TITLE
Next 13 adding <link rel="preload"> to images not using the images component - bad for LCP

### DISCUSSION THREAD

**Discussion by valley-boy** [vercel/next.js#54799] [Help]
### Summary

For various reasons (see below example) we don't want to use the next/image component for our images (we just want to use a regular img tag), however on our SSR pages Next is still setting a link rel="preload" tag in the head for each img without a loading=lazy. We don't want to set a loading = lazy, on these specific images, is there any other way we can prevent next from adding the link rel="preload" for images 

### Additional information

```js
The reason this killing our LCP that its selecting the siteimages/image-full.jpg (our high res fallback) as the <link rel="preload"> but its not used as most browsers support the webp. So we are loading x2 images when only 1 required!

<picture>
    // load webp in different sizes if browser supports it
    <source type="image/webp"
        sizes=" (max-width: 500px) 500px,
                (max-width: 768px) 768px,
                (max-width: 1024px) 1024px,
                100vw"
        srcset="
          webp/image-500.webp 500w,
          webp/image-768.webp 768w,
          webp/image-1024.webp 1024w,
          webp/image-full.webp 1920w
        "
        >
    // fallback on original src.
    <img 
        srcset="siteimages/image-1024.jpg 1024w, 
            siteimages/image-768.jpg 768w,
            siteimages/image-500.jpg 500w"
        sizes=" (max-width: 500px) 500px,
                (max-width: 768px) 768px,
                (max-width: 1024px) 1024px,
                100vw"
        src="siteimages/image-full.jpg" 

        alt="image description"
        class="myimageclass">
</picture>
```

**Comment by gnoff** [comment]
@valley-boy  agreed, this should be handled upstream in React correctly. We will update it so that images inside pictures are not automatically preloaded

**Comment by valley-boy** [comment]
Hi Josh, many thanks for the reply

If you need a working example on the following the first Nursing Home
listing preloads both the jpg fallback in the img src and and webp photo in
the source srcset, we ideally don't want the jpg preloading as it undoes
all our good performance work of trying to go webp vs jpg!

https://www.nursinghomes.com/il/chicago/ (our first next.js site and very
much enjoying it!)

look forward to the resolution, and very much appreciated

cheers

Oli

**Reply by gnoff** [reply]
preloading pictures is hard given there are multiple layers of selection and the browser is going to only pick one source to load. We could try to preload the first source inside a picture and for browsers that would fallback they'd just not benefit from the preload while browsers that do match the first source would get the preload. I'll work on this in the next week or two

**Comment by valley-boy** [comment]
Hi Josh understood, issue with loading the first source is that it may be
the right format for the browser but the wrong image size for current device

Maybe a safer option would be not to set any preload and leave it up to us
to add a preload link. We would then add a preload link with a srcset with
only the webp files, not the fallback jpg.  Preload supports the srcset so
works nicely.

cheers

Oli

**Reply by gnoff** [reply]
Hi Oli,

using `<link rel="preload" as="image" imagesrcset="...the source srcset" imagesizes="...the source sizes" type="...the source type" />` the browser should preload the correct size and only if it can support the image type. Are you saying that there are still cases where a preload constructed this way would potentially load the wrong size?

**Comment by valley-boy** [comment]
Hi Josh, sorry I see now, think this will work well,

 so in the above it would preload this part as the preload srcset

        type="image/webp"

        sizes=" (max-width: 500px) 500px,                (max-width:
768px) 768px,                (max-width: 1024px) 1024px,
 100vw"
        srcset="          webp/image-500.webp 500w,
webp/image-768.webp 768w,          webp/image-1024.webp 1024w,
 webp/image-full.webp 1920w        "

**Reply by gnoff** [reply]
Yeah so if the browser did not support webp it would not preload anything but for browsers that do support webp it would preload the right src from srcset based on the sizes defined so there should be no duplicate fetch for the image

**Comment by valley-boy** [comment]
sounds great, cheers Josh

**Reply by gnoff** [reply]
@valley-boy so upon further considering this is going to be more of challenge to implement ideally because you can suspend inside the `<picture>` and that can lead to incorrect behavior when determining which source would be approrpiate. I'm going to land a quick fix which just omits the preloads for img-inside-picture and consider a more comprehensive solution later

**Reply by gnoff** [reply]
fixed in https://github.com/facebook/react/pull/27346

**Comment by valley-boy** [comment]
cheers Josh, appreciated

Any idea of a timeline for the next stable release that will have this
amendment included?

**Reply by gnoff** [reply]
I expect sometime next week

**Comment by vankatadam** [comment]
@gnoff We have <noscript> part in our codebase, which before next13 app directory was working correctly. Now we are switching to server component and our noscript images are being preloaded even if js is on. Is this already a known Bug? And do you know if its Next or React related, so we could open the Issue in the correct forum.

Example:
```html
  <noscript>
    <img style={props?.style} alt={props.alt} src={props.src} />
  </noscript> 
```

**Reply by ascorbic** [reply]
I think a workaround for now is to add `loading="lazy"`

**Comment by vankatadam** [comment]
@ascorbic That is only a bad workaround for us, because we don't want to have all our images to be lazily loaded. @gnoff Is this already a known issue and is there a fix coming for this?

**Reply by ascorbic** [reply]
https://github.com/facebook/react/pull/27346

**Comment by yiminghe** [comment]
same issue in next 15.3.1

**Comment by mpkelly** [comment]
Sibling selectors are another reason there should be an opt-out. I have some rules that target sections that immediately follow sections with certain features, and those are broken in some cases due to preloads being plonked into the DOM.  


<img width="318" height="80" alt="Screenshot 2568-08-08 at 16 07 04" src="https://github.com/user-attachments/assets/0d8b768f-072a-410b-8ff1-dc71db22431a" />
